drivers/overlay/overlay.go

```diff
@@ -882,6 +882,32 @@ func (d *Driver) UpdateLayerIDMap(id string, toContainer, toHost *idtools.IDMapp
 	if err := idtools.MkdirAs(diffDir, 0755, rootUID, rootGID); err != nil {
 		return err
 	}
+
+	if rootUID != 0 {
+		if err = os.Chown(dir, rootUID, rootGID); err != nil {
+			return errors.Wrapf(err, "cannot chown '%s'", dir)
+		}
+		if err = os.Chmod(dir, 0700); err != nil {
+			return errors.Wrapf(err, "cannot chmod '%s'", dir)
+		}
+		parent := filepath.Dir(dir)
+		for parent != "/" {
+			fi, err := os.Stat(parent)
+			if err != nil {
+				return errors.Wrapf(err, "cannot stat '%s'", parent)
+			}
+
+			// An unprivileged user cannot still reach this directory as the parent
+			// has mode=0700
+			const desiredMask = 0755
```

Comment:
I had to change that as otherwise it would break the use case @nalind showed.

At the same time I made the parent directory `0700` so that will avoid any other user to look/read into this directory

Reply:
double checking it...

Reply:
true, I don't need `0755` here.  What I needed was the diffDir to have `0755`.

Pushed another patch to restore `0711` for the parent directories